django/db/models/fields/__init__.py

```diff
@@ -772,6 +769,8 @@ def get_db_prep_lookup(self, lookup_type, value, connection,
             return [self.get_db_prep_value(value, connection=connection,
                                            prepared=prepared)]
         elif lookup_type in ('range', 'in'):
+            if not isinstance(value, (list, tuple)):
```

Comment:
I'm not sure about this change. Why do we suddenly need to support non-iterables passed to `range` or `in` lookups in this PR? This could break code that currently works, if someone is passing an instance of a custom iterable class to `range` or `in`.